### PR TITLE
WIP do not merge. buy function cannot be front run. Cannot lose deposit accidently by c…

### PR DESCRIPTION
**Pull request by moose-code**
…hanging benefactor to yourself. Buidler added.

npx buidler test

Test still failing and need to reflect new buy function and the different buidler runner.

### CODE REVIEW

**Comment by JasoonS** [comment]
Cool, buidler is definitely a lower priority btw.
We can also make these PRs into `tweaks-and-fixes` rather than master, then at the end do the PR into master; like this PR: https://github.com/wild-cards/contracts/pull/36

**Comment by moose-code** [comment]
Screen Shot 2020-04-29 at 11 49 44

**Comment by moose-code** [comment]
@JasoonS Think this bit is done now

File: test/steward_admin_change.js [hunk 1]
```diff
@@ -4,15 +4,15 @@ const {
   ether,
   expectEvent,
   balance,
-  time
+  time,
```
Comment (JasoonS) [inline test/steward_admin_change.js:7]: These trailing commas caused some merge conflicts, because it changed code all over the place, but no stress.

**Pull request closed without merging**